test/e2e/v4/plugin_cluster_test.go

```diff
@@ -226,12 +231,19 @@ func Run(kbc *utils.TestContext, hasWebhook, isToUseInstaller bool) {
 		return err
 	}, time.Minute, time.Second).Should(Succeed())
 
-	By("checking the metrics values to validate that the created resource object gets reconciled")
-	metricsOutput := getMetricsOutput(kbc)
-	ExpectWithOffset(1, metricsOutput).To(ContainSubstring(fmt.Sprintf(
-		`controller_runtime_reconcile_total{controller="%s",result="success"} 1`,
-		strings.ToLower(kbc.Kind),
-	)))
+	if hasMetrics {
+		By("checking the metrics values to validate that the created resource object gets reconciled")
+		metricsOutput := getMetricsOutput(kbc)
+		ExpectWithOffset(1, metricsOutput).To(ContainSubstring(fmt.Sprintf(
+			`controller_runtime_reconcile_total{controller="%s",result="success"} 1`,
+			strings.ToLower(kbc.Kind),
+		)))
+	}
+
+	if !hasMetrics {
+		By("validating the metrics endpoint is not working as expected")
+		metricsShouldBeUnavailable(kbc)
+	}
```

Comment:
I would like to keep the test for no metrics so that we can ensure that users are able to disable the metrics.